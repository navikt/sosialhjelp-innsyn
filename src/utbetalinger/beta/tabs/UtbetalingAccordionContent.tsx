import { useTranslation } from "next-i18next";
import { Accordion, BodyShort } from "@navikt/ds-react";
import Link from "next/link";
import { FileTextIcon } from "@navikt/aksel-icons";

import type { ManedUtbetaling } from "../../../generated/model";
import { formatDato } from "../../../utils/formatting";
import { utbetalingsmetodeText } from "../../utbetalingerUtils";
import { logButtonOrLinkClick } from "../../../utils/amplitude";

export const UtbetalingAccordionContent = ({
    fom,
    tom,
    mottaker,
    annenMottaker,
    utbetalingsmetode,
    kontonummer,
    fiksDigisosId,
}: Pick<
    ManedUtbetaling,
    "fom" | "tom" | "mottaker" | "annenMottaker" | "utbetalingsmetode" | "kontonummer" | "fiksDigisosId"
>) => {
    const { t, i18n } = useTranslation("utbetalinger");

    return (
        <Accordion.Content className="pt-2">
            {fom && tom && (
                <>
                    <BodyShort className="font-bold">{t("periode")}</BodyShort>
                    <BodyShort spacing>
                        {formatDato(fom, i18n.language)} - {formatDato(tom, i18n.language)}
                    </BodyShort>
                </>
            )}
            <BodyShort className="font-bold">{t("mottaker")}</BodyShort>
            {annenMottaker ? (
                <BodyShort className="capitalize" spacing>
                    {mottaker}
                </BodyShort>
            ) : (
                <BodyShort spacing>
                    {t("tilDeg")} {utbetalingsmetodeText(utbetalingsmetode, i18n)} {kontonummer}
                </BodyShort>
            )}

            <Link
                href={`/${fiksDigisosId}/status`}
                className="navds-link items-center gap-2 flex"
                onClick={() => logButtonOrLinkClick("Åpner søknaden fra utbetalingen")}
            >
                <FileTextIcon aria-hidden width="1.5rem" height="1.5rem" />
                {t("soknadLenke")}
            </Link>
        </Accordion.Content>
    );
};
