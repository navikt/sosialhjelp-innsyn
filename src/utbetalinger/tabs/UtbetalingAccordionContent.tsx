import { useTranslations } from "next-intl";
import { Accordion, BodyShort, Link as AkselLink } from "@navikt/ds-react";
import Link from "next/link";
import { FileTextIcon } from "@navikt/aksel-icons";

import type { ManedUtbetaling } from "../../generated/model";
import { formatDato } from "../../utils/formatting";
import { logButtonOrLinkClick } from "../../utils/amplitude";

export const UtbetalingAccordionContent = ({
    fom,
    tom,
    mottaker,
    annenMottaker,
    utbetalingsmetode,
    kontonummer,
    fiksDigisosId,
}: Pick<
    ManedUtbetaling,
    "fom" | "tom" | "mottaker" | "annenMottaker" | "utbetalingsmetode" | "kontonummer" | "fiksDigisosId"
>) => {
    const { t, i18n } = useTranslation("utbetalinger");

    const utbetalingsmetodeTekst = !utbetalingsmetode
        ? null
        : i18n.exists(`utbetalingsmetode.${utbetalingsmetode?.toLowerCase()}`)
          ? i18n.t(`utbetalingsmetode.${utbetalingsmetode?.toLowerCase()}`)
          : utbetalingsmetode;

    return (
        <Accordion.Content className="pt-2">
            {fom && tom && (
                <>
                    <BodyShort className="font-bold">{t("periode")}</BodyShort>
                    <BodyShort spacing>
                        {formatDato(fom, i18n.language)} - {formatDato(tom, i18n.language)}
                    </BodyShort>
                </>
            )}
            <BodyShort className="font-bold">{t("mottaker")}</BodyShort>
            {annenMottaker ? (
                <BodyShort className="capitalize" spacing>
                    {mottaker}
                </BodyShort>
            ) : (
                <BodyShort spacing>
                    {t("tilDeg")} {utbetalingsmetodeTekst} {kontonummer}
                </BodyShort>
            )}
            <AkselLink
                as={Link}
                href={`/${fiksDigisosId}/status`}
                className="navds-link items-center gap-2 flex"
                onClick={() => logButtonOrLinkClick("Åpner søknaden fra utbetalingen")}
            >
                <FileTextIcon aria-hidden width="1.5rem" height="1.5rem" />
                {t("soknadLenke")}
            </AkselLink>
        </Accordion.Content>
    );
};
