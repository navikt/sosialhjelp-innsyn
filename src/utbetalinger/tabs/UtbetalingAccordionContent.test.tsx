import { render } from "@testing-library/react";
import { Accordion } from "@navikt/ds-react";

import { UtbetalingAccordionContent } from "./UtbetalingAccordionContent";

describe("UtbetalingAccordionContent", () => {
    it("matches snapshot with annenMottaker", () =>
        expect(
            render(
                <Accordion>
                    <Accordion.Item>
                        <UtbetalingAccordionContent
                            fom="2022-01-01"
                            tom="2022-01-31"
                            mottaker="Ola Nordmann"
                            annenMottaker={true}
                            utbetalingsmetode="BANK"
                            kontonummer="1234 56 78901"
                            fiksDigisosId="123"
                        />
                    </Accordion.Item>
                </Accordion>
            ).asFragment()
        ).toMatchSnapshot());
});
