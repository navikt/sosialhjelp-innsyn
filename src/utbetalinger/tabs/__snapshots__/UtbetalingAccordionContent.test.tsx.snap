// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UtbetalingAccordionContent > matches snapshot with annenMottaker 1`] = `
<DocumentFragment>
  <div
    class="navds-accordion navds-accordion--medium navds-accordion--indent"
  >
    <div
      class="navds-accordion__item"
      data-expanded="false"
    >
      <div
        aria-hidden="true"
        class="navds-accordion__content navds-accordion__content--closed pt-2 navds-body-long navds-body-long--medium"
      >
        <p
          class="font-bold navds-body-short navds-body-short--medium"
        >
          Periode
        </p>
        <p
          class="navds-body-short navds-body-short--medium navds-typo--spacing"
        >
          1. januar 2022 - 31. januar 2022
        </p>
        <p
          class="font-bold navds-body-short navds-body-short--medium"
        >
          Mottaker
        </p>
        <p
          class="capitalize navds-body-short navds-body-short--medium navds-typo--spacing"
        >
          Ola Nordmann
        </p>
        <a
          class="navds-link navds-link items-center gap-2 flex navds-link--action"
          data-color="accent"
          data-variant="action"
          href="/123/status"
        >
          <svg
            aria-hidden="true"
            fill="none"
            focusable="false"
            height="1.5rem"
            role="img"
            viewBox="0 0 24 24"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M6.5 3.25c-.69 0-1.25.56-1.25 1.25v15c0 .69.56 1.25 1.25 1.25h11c.69 0 1.25-.56 1.25-1.25V8a.75.75 0 0 0-.22-.53l-4-4a.75.75 0 0 0-.53-.22zm6.75 4.25c0 .69.56 1.25 1.25 1.25h2.75v10.5H6.75V4.75h6.5zm2.94-.25-1.44-1.44v1.44zM8.25 13a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 0 1.5H9a.75.75 0 0 1-.75-.75M9 15.25a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          Gå til søknaden og les vedtaksbrevet for mer detaljer
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
